runtime/vm/ModularityHashTables.c

```diff
@@ -24,21 +24,62 @@
 #include "j9protos.h"
 #include "ut_j9vm.h"
 
+static j9object_t moduleHashGetName(const void *entry);
+static UDATA moduleNameHashFn(void *key, void *userData);
+static UDATA moduleNameHashEqualFn(void *leftKey, void *rightKey, void *userData);
+static UDATA modulePointerHashFn(void *key, void *userData);
+static UDATA modulePointerHashEqualFn(void *leftKey, void *rightKey, void *userData);
 static UDATA packageHashFn(void *key, void *userData);
 static UDATA packageHashEqualFn(void *leftKey, void *rightKey, void *userData);
-static UDATA moduleHashFn(void *key, void *userData);
-static UDATA moduleHashEqualFn(void *leftKey, void *rightKey, void *userData);
 static UDATA moduleExtraInfoHashFn(void *key, void *userData);
 static UDATA moduleExtraInfoHashEqualFn(void *tableNode, void *queryNode, void *userData);
 
+static j9object_t
+moduleHashGetName(const void *entry)
+{
+	const J9Module** const modulePtr = (const J9Module**)entry;
+	const J9Module* const module = *modulePtr;
+	j9object_t moduleName = module->moduleName;
+
+	return moduleName;
+}
+static UDATA 
+moduleNameHashFn(void *key, void *userData) 
+{
+	J9JavaVM *javaVM = (J9JavaVM*) userData;
+	j9object_t name = moduleHashGetName(key);
+
+	return javaVM->memoryManagerFunctions->j9gc_stringHashFn(&name, userData);
```

Comment:
Since the classLoader is included in the equals function, it should affect the hash as well.

Reply:
`moduleNameHashEqualFn` and `moduleNameHashFn` are for `classLoader->moduleHashTable` in which all modules should have same `classLoader`. The comparison of `(tableNodeModule->classLoader == queryNodeModule->classLoader)` within `moduleNameHashEqualFn` can be removed or become `Assert`.

Reply:
Changed the comparison `(tableNodeModule->classLoader == queryNodeModule->classLoader)` to an `Assert`, please have another look.